src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -1245,6 +1245,11 @@ internal BoundExpression NullOrDefault(TypeSymbol typeSymbol)
 #endif
             )
         {
+            if (refKind == RefKind.Out)
```

Comment:
As I found it is very convenient when creating a temp for a parameter - you just pass the ref kind of the parameter. 
I did not want to change all that code, just intercept "Out" in the helper.